WebApi/Controllers/BooksController.cs

```diff
@@ -0,0 +1,261 @@
+        /// Adding genre to the book
+        /// </summary>
+        /// <param name="bookId">Book id for genre add</param>
+        /// <param name="genreId">Genre id for add</param>
+        /// <returns>Status of adding genre</returns>
+        [HttpPost("{bookId}/genres/{genreId}")]
+        public IActionResult AddBookGenre(int bookId, int genreId)
+        {
+            if (this.library.AddBookGenre(bookId, genreId))
+            {
+                return this.Ok();
+            }
+
+            return this.BadRequest("Book or genre doesn`t exist or book already has this genre");
+        }
+
+        /// <summary>
+        /// Getting all books method
+        /// </summary>
+        /// <returns>List of books</returns>
+        [HttpGet]
+        public IActionResult GetAllBooks()
+        {
+            return this.Ok(this.library.GetAllBooks().ToList());
+        }
+
+        /// <summary>
+        /// Getting single book by id
+        /// </summary>
+        /// <param name="id">Book id to get</param>
+        /// <returns>Book or NotFound</returns>
+        [HttpGet("{id}")]
+        public IActionResult GetBookById(int id)
+        {
+            Book resultBook = this.library.GetBookById(id);
+            if (resultBook == null)
+            {
+                return this.NotFound();
+            }
+
+            return this.Ok(resultBook);
+        }
+
+        /// <summary>
+        /// Getting book authors
+        /// </summary>
+        /// <param name="bookId">Book id to get authors</param>
+        /// <returns>List of authors</returns>
+        [HttpGet("{bookId}/authors")]
+        public IActionResult GetBookAuthors(int bookId)
+        {
+            List<Author> authors = this.library.GetBookAuthors(bookId);
+            if (authors != null && authors.Count != 0)
+            {
+                return this.Ok(authors);
+            }
+
+            return this.NotFound("Book doesn`t exist or book doesn`t containts authors!");
+        }
+
+        /// <summary>
+        /// Getting book genres
+        /// </summary>
+        /// <param name="bookId">Book id to get genres</param>
+        /// <returns>List of genres</returns>
+        [HttpGet("{bookId}/genres")]
+        public IActionResult GetBookGenres(int bookId)
+        {
+            List<Genre> genres = this.library.GetBookGenres(bookId);
+            if (genres != null && genres.Count != 0)
+            {
+                return this.Ok(genres);
+            }
+
+            return this.NotFound("Book doesn`t exist or book doesn`t containts authors!");
+        }
+
+        /// <summary>
+        /// Getting books of genre
+        /// </summary>
+        /// <param name="genreId">Genre id to get its books</param>
+        /// <returns>List of books</returns>
+        [HttpGet("{genreId}/books")]
+        public IActionResult GetBooksOfGenre(int genreId)
+        {
+            List<Book> booksOfGenre = this.library.GetBooksInGenre(genreId).ToList();
```

Comment:
Same thing with naming here. As you don't have any other list of books in this method. And this method called GetBooksOfGenre and you are calling this.library.GetBooksInGenre it will be enough to call this variable just 'books'.
But it's more your chose.